tools/path-encoding-verify.py

```
@@ -0,0 +1,123 @@
+            check(alt != vector["encoded"], name, f"alternate {alt!r} repeats the reference encoded form")
+            check(ALT_SEGMENT.fullmatch(alt) is not None, name, f"alternate {alt!r} has a raw reserved character or bad %HH")
+            check(decodes_to(alt, key), name, f"alternate {alt!r} does not decode to key")
+    return len(document["vectors"])
+
+
+def self_test(document: dict) -> None:
+    """Each poisoned copy must trip a distinct guard — otherwise verify() is toothless."""
+    def row(vectors: list, key: str) -> dict:
+        return next(v for v in vectors if v["key"] == key)
+
+    def set_field(key: str, field: str, value: object):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `set_field`

(ANN202)